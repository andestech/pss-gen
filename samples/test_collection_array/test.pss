component pss_top {
	action root_a {
		array <int, 3> a = {4, 5, 6};
		array <int, 3> b;
		array <int, 3> c;
		array <int, 3> d = {10, 11, 12};
		array <int, 3> e = {11, 10, 12};

		exec pre_solve {
			b[0] = 7;
			b[1] = 8;
			b[2] = 9;
			c = {10, 11, 12};
		}

		activity {
			do sub_a with {
				a[0] == this.a[0];
				a[1] == this.a[1];
				a[2] == this.a[2];
			};
			do sub_a with {
				a[0] == b[0];
				a[1] == b[1];
				a[2] == b[2];
			};
			do sub_a with {
				a[0] == c[0];
				a[1] == c[1];
				a[2] == c[2];
			};
            if (c == d)
                do printf with {val == 1;};
            if (c != e)
                do printf with {val == 1;};
		}
	}

	action printf {
		rand int val;
		exec body ASM =
		"""val = {{val}}""";
	}

	action sub_a {
		rand int a[3];
		exec body ASM = """a[0]={{a[0]}}, a[1]={{a[1]}}, a[2]={{a[2]}}""";

	}
}

