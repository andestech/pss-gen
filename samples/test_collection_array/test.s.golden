a[0]=4, a[1]=5, a[2]=6
a[0]=7, a[1]=8, a[2]=9
a[0]=10, a[1]=11, a[2]=12
val = 1
val = 1
val = 10
val = 11
val = 12
